src/TesterInternal/General/BuiltInSerializerTests.cs

```diff
@@ -29,7 +29,7 @@ public class BuiltInSerializerTests
         public void InitializeForTesting()
         {
             TraceLogger.Initialize(new NodeConfiguration());
-            SerializationManager.Initialize(false, null);
+            SerializationManager.Initialize(false, null, false);
```

Comment:
OK, now I also remember why I used object and not T in the first place. Different commands return results of different types, so cannot use the same T for all.

OK, so lets just then instead of doing `Parse` do:
`int[] numAgents = results.Select(r => Convert.ToInt32(r)).ToArray();`
It will be cleaner I think. The `parse` confused me that json some how does not know how to deserialize ints.
Also, worth putting a comment next to this code with the link that you have shared here (http://stackoverflow.com/a/17918824) and explain this issue with boxing.